app/controllers/feedback/Survey.scala

```diff
@@ -52,13 +53,23 @@ trait PostSubmitFeedback extends FrontendController {
 
   def formCompleteFeedback = RefNumAction.async { implicit request =>
     completedFeedbackForm.bindFromRequest.fold(
-      formWithErrors => viewConfirmationPage(request.refNum, formWithErrors),
+      formWithErrors => BadRequest(views.html.inpageAfterSubmissionFeedbackForm(formWithErrors)),
       success => {
         sendFeedback(success, request.refNum) map { _ => Redirect(routes.Survey.surveyThankyou()) }
       }
     )
   }
 
+  def inpageAfterSubmissionFeedbackForm  = RefNumAction { implicit request =>
+
+    val completedFeedbackForm = Form(mapping(
```

Comment:
you can get rid of this form since it's literally identical to the one on line 45